solidity/contracts/bridge/Fraud.sol

```diff
@@ -226,7 +227,10 @@ library Fraud {
 
         // Check that the UTXO key identifies a correctly spent UTXO.
         require(
-            self.deposits[utxoKey].sweptAt > 0 || self.spentMainUTXOs[utxoKey],
+            self.deposits[utxoKey].sweptAt > 0 ||
+                self.spentMainUTXOs[utxoKey] ||
+                self.movedFundsSweepRequests[utxoKey].state ==
+                MovingFunds.MovedFundsSweepRequestState.Processed,
```

Comment:
This will work, but as a simpler alternative we could consider renaming `spentMainUtxos` to `provedUtxos` (or something like that) and put there both of the inputs of the "funds moved" transaction.

`provedUTXO` could hold all the inputs proved as correctly spent from the `submit*Proof` functions.

(Just a suggestion, it can stay as it is).

Reply:
That will cost more gas in `submitMovedFundsSweepProof`. Now we just change the `state` of an existing request but in the proposed approach we'll need to put a new entry to the `provedUtxos` mapping. Also, I think it's cleaner to have the current approach where things are separated. We need to hold `movedFundsSweepRequests` mapping anyway.